src/dataServices/dynamoDbDataService.ts

```diff
@@ -385,6 +385,18 @@ export class DynamoDbDataService implements Persistence, BulkDataAccess {
         return exportJob;
     }
 
+    async getActiveSubscriptions(params: { tenantId?: string }): Promise<Record<string, any>[]> {
+        this.assertValidTenancyMode(params.tenantId);
+        const subscriptionQuery = DynamoDbParamBuilder.buildGetActiveSubscriptions(params.tenantId);
+        const queryResponse = await this.dynamoDb.query(subscriptionQuery).promise();
+        const subscriptions: Record<string, any>[] = [];
+        queryResponse.Items?.forEach((response) => {
+            const item = DynamoDBConverter.unmarshall(response);
+            subscriptions.push({[item.id]: item});
+        });
+        return subscriptions;
```

Comment:
consider the case where you need to paginate through the query results (i.e. `LastEvaluatedKey` is present)

Something like https://github.com/carvantes/fhir-works-on-aws-deployment/blob/c7c63417e655e66ee0d7ef5d8c84b2d50ee4f8ad/scripts/sort-key-migration.js#L82

Reply:
Unfortunately using the parambuilder to generate the query meant that it wasn't easy to add the ExclusiveStartKey, so I had to use the defineProperty function.